odhcp6c: update to Git HEAD (2025-02-06) and conditionally request S46 OROs

Headcrabed opened the pull request:
8aa8b706727a Use GNUInstallDirs macros
This bump needs to be merged together with https://github.com/openwrt/openwrt/pull/18351

Edit: now this PR also includes changes from #18351 since original author does not reply.
Closes #14056

hauke commented:
This also includes this commit: https://github.com/openwrt/odhcp6c/commit/ffbb2d559af42ddb38f315ce0f5f9f70204e1037 There is a discussion about it in #16209.

Neustradamus commented:
@Headcrabed: Nice PR!

Headcrabed commented:
@hauke Since #18351 's owner does not reply at all, I took that commit and include that into my PR. Can you review?

Headcrabed commented:
@hauke The test case you have triggered are all successful.

hauke commented:
Did you test this change? Do you see the DHCP options when you should see them?

Headcrabed committed:
odhcp6c: Conditionally request S46 OROs

Request Softwire46 (S46) [RFC 7598] options when the map and/or ds-lite
packages are installed. This is required as the behaviour of odhcp6c has
changed to not include these OROs by default.

See openwrt/odhcp6c#89

Signed-off-by: Richard Patterson <richard@helix.net.nz>
Signed-off-by: Shengyu Qu <wiagn233@outlook.com>

Headcrabed committed:
odhcp6c: update to Git HEAD (2025-02-06)

ffbb2d559af4 odhcp6c: Do not request S46 OROs by default
8aa8b706727a Use GNUInstallDirs macros

Signed-off-by: Shengyu Qu <wiagn233@outlook.com>

Headcrabed commented:
> Did you test this change? Do you see the DHCP options when you should see them?

`odhcp6c: Conditionally request S46 OROs` is tested working. However, there's something weird about `PKG_MIRROR_HASH`. 

Using [this](https://git.openwrt.org/?p=maintainer-tools.git;a=blob;f=update_git_source_package.sh;h=d7f823755d6ed7573aa15f703859f01a9017b016;hb=HEAD) script gives different hash value between [this method](https://openwrt.org/docs/guide-developer/packages#use_source_repository). Which hash value should I use?

Headcrabed commented:
> > Did you test this change? Do you see the DHCP options when you should see them?
> 
> @hauke `odhcp6c: Conditionally request S46 OROs` is tested working. However, there's something weird about `PKG_MIRROR_HASH`.
> 
> Using [this](https://git.openwrt.org/?p=maintainer-tools.git;a=blob;f=update_git_source_package.sh;h=d7f823755d6ed7573aa15f703859f01a9017b016;hb=HEAD) script gives different hash value between [this method](https://openwrt.org/docs/guide-developer/packages#use_source_repository). Which hash value should I use?

@hauke I found the reason for different hash value. That's due to different `PKG_SOURCE_DATE` value. Now everything is tested and I think it's ready now.

Headcrabed commented:
@hauke Ping

ldir-EDB0 commented:
MAP-T / nftables needs to be resolved as well, otherwise that's at least 2 countries where openwrt is feature incomplete.

Headcrabed commented:
@hauke Ping again.
